src/targets/native/canvas.tsx

```diff
@@ -1,93 +1,105 @@
 import * as React from 'react'
-import { GLView } from 'expo-gl'
-import { LayoutChangeEvent, PixelRatio, ViewStyle, PanResponder, GestureResponderEvent } from 'react-native'
+import { GLView, ExpoWebGLRenderingContext } from 'expo-gl'
+import {
+  View,
+  LayoutChangeEvent,
+  PixelRatio,
+  ViewStyle,
+  PanResponder,
+  GestureResponderEvent,
+  StyleSheet,
+} from 'react-native'
 import { Renderer } from 'expo-three'
-import { useEffect, useState } from 'react'
-import { useCanvas, CanvasProps } from '../../canvas'
+import { useState } from 'react'
+import { useCanvas, CanvasProps, RectReadOnly } from '../../canvas'
 
 function clientXY(e: GestureResponderEvent) {
   ;(e as any).clientX = e.nativeEvent.pageX
   ;(e as any).clientY = e.nativeEvent.pageY
   return e
 }
 
-export const Canvas = React.memo((props: Omit<CanvasProps, 'style'> & { style?: ViewStyle }) => {
-  const [gl, setGl] = useState()
-  const [glContext, setGlContext] = useState()
+type NativeCanvasProps = Omit<CanvasProps, 'style'> & {
+  style?: ViewStyle
+  onContextCreated?: (gl: ExpoWebGLRenderingContext) => Promise<any> | void
+}
 
-  const [pixelRatio, setPixelRatio] = useState(props.pixelRatio || 1)
-  const [size, setSize] = useState({ width: 0, height: 0, top: 0, left: 0 })
+const styles: ViewStyle = { flex: 1 }
 
-  const { pointerEvents } = useCanvas({
-    ...props,
-    size,
-    pixelRatio,
-    gl,
-  })
+const IsReady = React.memo(({ gl, ...props }: NativeCanvasProps & { gl: any; size: any }) => {
+  const { pointerEvents } = useCanvas({ ...props, gl, browser: true })
```

Comment:
@birkir it says browser: true, was this intended?

Reply:
Oh snap, no it should be false. But it seems to me that it works either way... I was under the impression that useLayoutEffect is an DOM only api? As it gets triggered after layout changes

Should we go ahead and remove the flag all together? I don't know the implications of using it in native or any other targets.

Reply:
sure, if rn has that we can remove it. there's one more place in canvas where it's being used

```jsx
if (browser && event.target && event.target.offsetParent) {
```

but just checking for even.target and offsetparent will be enough i guess.